rust-runtime/aws-smithy-http-server/src/operation/upgrade.rs

```diff
@@ -255,17 +254,20 @@ where
     // The signature of the output is correct
     <Pl::Layer as Layer<Upgrade<P, Op, Exts, B, Pl::Service>>>::Service:
         Service<http::Request<B>, Response = http::Response<BoxBody>>,
-{
-    type Service = <Pl::Layer as Layer<Upgrade<P, Op, Exts, B, Pl::Service>>>::Service;
 
+    // For `Route::new` for the resulting service
+    <Pl::Layer as Layer<Upgrade<P, Op, Exts, B, Pl::Service>>>::Service: tower::Service<http::Request<B>, Error = std::convert::Infallible>,
```

Comment:
We have `tower::Service` and `Infallible` imported here so we don't need `tower::Service` and `std::convert::Infallible`.

Same for line 261.

Reply:
It might be worth adding a type alias to reduce the length of these.